cmd/searcher/search/search_structural_test.go

```diff
@@ -13,6 +13,75 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/testutil"
 )
 
+func TestMatcherLookupByLanguage(t *testing.T) {
+	// If we are not on CI skip the test.
+	if os.Getenv("CI") == "" {
+		t.Skip("Not on CI, skipping comby-dependent test")
+	}
+
+	input := map[string]string{
+		"file_without_extension": `
+/* This foo(plain string) {} is in a Go comment should not match in Go, but should match in plaintext */
+func foo(go string) {}
+`,
+	}
+
+	p := &protocol.PatternInfo{
+		Pattern:         "foo(:[args])",
+		IncludePatterns: []string{"file_without_extension"},
+	}
+
+	cases := []struct {
+		Name      string
+		Languages []string
+		Want      []string
+	}{
+		{
+			Name:      "Language test for Go",
+			Languages: []string{"go"},
+			Want:      []string{"foo(go string)"},
+		},
+		{
+			Name:      "Language test for plaintext",
+			Languages: []string{"text"},
+			Want:      []string{"foo(plain string)", "foo(go string)"},
+		},
+	}
+
+	zipData, err := testutil.CreateZip(input)
+	if err != nil {
+		t.Fatal(err)
+	}
+	zf, cleanup, err := testutil.TempZipFileOnDisk(zipData)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+
+	for _, tt := range cases {
+		t.Run(tt.Name, func(t *testing.T) {
+			p.Languages = tt.Languages
+			matches, _, err := structuralSearch(context.Background(), zf, p.Pattern, p.CombyRule, p.Languages, p.IncludePatterns, "repo_foo")
+			if err != nil {
+				t.Fatal(err)
+			}
+			var got []string
+			for _, fileMatches := range matches {
+				for _, m := range fileMatches.LineMatches {
+					got = append(got, m.Preview)
+				}
+			}
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !reflect.DeepEqual(got, tt.Want) {
+				t.Fatalf("got file matches %v, want %v", got, tt.Want)
```

Comment:
With tests like these that don't depend on each other, I think it's nice to have the assertions be `Errorf` when possible so that you can run more test cases and get more output:

```suggestion
				t.Errorf("got file matches %v, want %v", got, tt.Want)
```

Reply:
Sweet. You gave me an idea for another campaign 😏 . I'm going to do this in a separate PR to cover everything.